### PR TITLE
fix(hil-logic): Updated to use latest shared library schemas

### PR DESCRIPTION
**Pull request by jaycosaur**

### CODE REVIEW

File: functions/src/dialogue/public/dialogueSendMessage.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ const defaultDeps = {
   sendMessageToBot,
 };
 
-export const dialogueSendMessage: EventTriggers.IHttpsCallEvent<Dialogue.ISendMessageParams> = async (
+export const dialogueSendMessage: EventTriggers.IHttpsCallEvent<Dialogue.ISendMessageAsUserParams> = async (
```
Comment (zaripych) [inline functions/src/dialogue/public/dialogueSendMessage.ts]: Can we please rename this to `dialogueSendMessageAsUser` or `dialogueSendOnUserMessagePubSub`

File: functions/src/dialogue/public/dialogueSendMessage.ts [hunk 2]
```diff
@@ -21,20 +21,20 @@ export const dialogueSendMessage: EventTriggers.IHttpsCallEvent<Dialogue.ISendMe
...
-  const params = validate(rawParams, Dialogue.sendMessageParamsSchema, 'public');
+  const params = validate(rawParams, Dialogue.sendMessageAsUserParamsSchema, 'public');
 
   const record: Models.IDbMessage = {
     message: {
       from: 'user',
-      text: params.message,
+      text: params.recognitionResult.text,
       timestamp: Date.now(),
       senderId: uid,
```
Comment (zaripych) [inline functions/src/dialogue/public/dialogueSendMessage.ts]: Do we need to set the `senderId` as `consoleId` here?

File: functions/src/dialogue/public/dialogueSendMessageAsHIL.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ const defaultDeps = {
   saveMessageToDbAndNotify,
 };
 
-export const dialogueSendMessageAsHIL: EventTriggers.IHttpsCallEvent<Dialogue.ISendMessageParams> = async (
+export const dialogueSendMessageAsHIL: EventTriggers.IHttpsCallEvent<Dialogue.ISendMessageAsHilParams> = async (
```
Comment (zaripych) [inline functions/src/dialogue/public/dialogueSendMessageAsHIL.ts]: `AsHil`

File: functions/src/dialogue/public/dialogueSendMessage.ts (hunk 1 above)
Comment (zaripych) [reply functions/src/dialogue/public/dialogueSendMessage.ts]: If this is a public function, please keep it as `dialogueSendMessageAsUser`, but I don't know why we would need one as public ...

File: functions/src/dialogue/public/dialogueSendMessageAsHIL.ts (hunk 3 above)
Comment (zaripych) [reply functions/src/dialogue/public/dialogueSendMessageAsHIL.ts]: And the file name please.

File: functions/src/sentience/private/handleMessage.ts [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+export const handleChatMessageOnTopic: EventTriggers.ISingleTopicEvent<Dialogue.ISendMessageAsUserParams> = async (
+  msg,
+  ctx
+) => {
+  const message: Dialogue.ISendMessageAsUserParams = msg.json;
 
-  return await sendMessageToBot(message);
+  console.log('Context Log: ', ctx);
+
+  if (ctx.auth) {
```
Comment (zaripych) [inline functions/src/sentience/private/handleMessage.ts]:
```
if (!ctx.auth) {
  return null;
}
```

File: functions/src/sentience/publishMessageToBot.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
-import { ISendMessageToBotParams } from './models';
 import { handleChatMessageOnTopic } from './private';
 import { publishToHandler } from '@shared/pubsub';
+import { Dialogue } from '@aimee-blue/ab-shared';
 
-export const publishMessageToBot = async (message: ISendMessageToBotParams) => {
+export const publishMessageToBot = async (message: Dialogue.ISendMessageAsUserParams) => {
```
Comment (zaripych) [inline functions/src/sentience/publishMessageToBot.ts]: `dialogueSendOnUserMessagePubSub`

Comment (zaripych) [reply functions/src/sentience/publishMessageToBot.ts]: And the file please.

File: functions/src/dialogue/public/dialogueSendMessage.ts (hunk 2 above)
Comment (jaycosaur) [reply functions/src/dialogue/public/dialogueSendMessage.ts]: Will be deleted.

**Pull request merged**